src/Events/QR/Redirections.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * The Redirections class for the QR module.
+ *
+ * @since TBD
+ */
+
+namespace TEC\Events\QR;
+
+use TEC\Common\Contracts\Provider\Controller;
+use TEC\Events\QR\Routes;
+use TEC\Events\QR\Settings;
+use Tribe__Events__Main as TEC;
+
+/**
+ * Class Redirections.
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\QR
+ */
+class Redirections extends Controller {
+	/**
+	 * Register the controller.
+	 *
+	 * @since TBD
+	 * @return void
+	 */
+	public function do_register(): void {
+		$this->add_hooks();
+	}
+
+	/**
+	 * Unregister the controller.
+	 *
+	 * @since TBD
+	 * @return void
+	 */
+	public function unregister(): void {
+		$this->remove_hooks();
+	}
+
+	/**
+	 * Adds the actions required by the controller.
+	 *
+	 * @since TBD
+	 * @return void
+	 */
+	protected function add_hooks(): void {
+		add_action( 'template_redirect', [ $this, 'handle_qr_redirect' ] );
+	}
+
+	/**
+	 * Removes the actions required by the controller.
+	 *
+	 * @since TBD
+	 * @return void
+	 */
+	protected function remove_hooks(): void {
+		remove_action( 'template_redirect', [ $this, 'handle_qr_redirect' ] );
+	}
```

Comment:
```suggestion
	public function do_register(): void {
		add_action( 'template_redirect', [ $this, 'handle_qr_redirect' ] );
	}

	/**
	 * Unregister the controller.
	 *
	 * @since TBD
	 * @return void
	 */
	public function unregister(): void {
		remove_action( 'template_redirect', [ $this, 'handle_qr_redirect' ] );
	}
```